src/keys/signature.rs

```diff
@@ -48,6 +48,14 @@ impl Signature {
             _ => None,
         }
     }
+
+    /// Returns ed25519_dalek::Signature if Self is a Ed25519 variant.
+    pub fn into_ed(self) -> Option<ed25519_dalek::Signature> {
```

Comment:
Wouldn't/shouldn't this be implemented with the `From` trait?: https://doc.rust-lang.org/rust-by-example/conversion/from_into.html

Reply:
We can do that for structs, but `Signature` is an enum here where it can be any one of the variants(BLS, Ed or BLSShare) so we have to `Option`-ize them. So we can't use the `From` trait here :)